handshake/crypto_setup_server_test.go

```diff
@@ -100,7 +101,11 @@ type mockStream struct {
 }
 
 func (s *mockStream) Read(p []byte) (int, error) {
-	return s.dataToRead.Read(p)
+	n, _ := s.dataToRead.Read(p)
+	if n == 0 { // block if there's no data
+		time.Sleep(time.Hour)
```

Comment:
If you need to block, block on something that you can unblock during test shutdown (e.g. a chan).